tests/validate_tests.rs

```diff
@@ -0,0 +1,512 @@
+    let contract: &BaseContract = context.opcodes_factory.contract().deref().deref();
+
+    let token = vec![Token::String(init_func)];
+    let function = contract.abi().function("create")?;
+    let init_func = function.encode_input(&token)?;
+    let mut init_code = vec![];
+    init_code.extend_from_slice(context.opcodes_factory.address.as_bytes());
+    init_code.extend_from_slice(&init_func);
+    Ok((Bytes::from(init_code), Bytes::from(init_func)))
+}
+
+async fn create_test_user_op(
+    validate_rule: String,
+    pm_rule: Option<String>,
+    init_code: Bytes,
+    init_func: Bytes,
+    factory_address: Address,
+) -> anyhow::Result<UserOperation> {
+    let context = get_global_context().await;
+
+    let paymaster_and_data = if let Some(rule) = pm_rule {
+        let mut data = vec![];
+        data.extend_from_slice(context.paymaster.address.as_bytes());
+        data.extend_from_slice(rule.as_bytes());
+        Bytes::from(data)
+    } else {
+        Bytes::default()
+    };
+
+    let signature = Bytes::from(validate_rule.as_bytes().to_vec());
+
+    let mut tx = TypedTransaction::default();
+    tx.set_to(factory_address);
+    tx.set_data(init_func);
+
+    let call_init_code_for_addr = context.client.call(&tx, None).await?;
+
+    let (head, address) = call_init_code_for_addr.split_at(12);
+    anyhow::ensure!(
+        !head.iter().any(|i| *i != 0),
+        format!(
+            "call init code returns non address data : {:?}",
+            call_init_code_for_addr
+        )
+    );
+
+    let sender = Address::from_slice(address);
+
+    Ok(UserOperation {
+        sender,
+        nonce: U256::zero(),
+        init_code,
+        call_data: Bytes::default(),
+        call_gas_limit: U256::from(1000000),
+        verification_gas_limit: U256::from(1000000),
+        pre_verification_gas: U256::from(50000),
+        max_fee_per_gas: U256::from(0),
+        max_priority_fee_per_gas: U256::from(0),
+        paymaster_and_data,
+        signature,
+    })
+}
+
+async fn existing_storage_account_user_op(validate_rule: String, pm_rule: String) -> UserOperation {
+    let context = get_global_context().await;
+
+    let mut paymaster_and_data = vec![];
+    paymaster_and_data.extend_from_slice(context.paymaster.address.as_bytes());
+    paymaster_and_data.extend_from_slice(pm_rule.as_bytes());
+
+    let signature = Bytes::from(validate_rule.as_bytes().to_vec());
+    UserOperation {
+        sender: context.storage_account.address,
+        nonce: U256::zero(),
+        init_code: Bytes::default(),
+        call_data: Bytes::default(),
+        call_gas_limit: U256::from(1000000),
+        verification_gas_limit: U256::from(1000000),
+        pre_verification_gas: U256::from(50000),
+        max_fee_per_gas: U256::from(0),
+        max_priority_fee_per_gas: U256::from(0),
+        paymaster_and_data: Bytes::from(paymaster_and_data),
+        signature,
+    }
+}
+
+fn validate(_user_op: UserOperation) -> anyhow::Result<()> {
```

Comment:
@Vid201 You have to implement this method for validations you wrote in your [pr](https://github.com/Vid201/aa-bundler/pull/24).